iocore/net/SSLUtils.cc

```diff
@@ -1053,8 +1053,8 @@ SSLPrivateKeyHandler(SSL_CTX *ctx, const SSLConfigParams *params, const char *ke
     scoped_BIO bio(BIO_new_mem_buf(secret_data, secret_data_len));
     pkey = PEM_read_bio_PrivateKey(bio.get(), nullptr, nullptr, nullptr);
     if (nullptr == pkey) {
-      Debug("ssl_load", "failed to load server private key from %s",
-            (!keyPath || keyPath[0] == '\0') ? "[empty key path]" : keyPath);
+      SSLError("failed to load server private key (%.*s) from %s", secret_data_len < 50 ? secret_data_len : 50, secret_data,
```

Comment:
I'm not sure when to use SSLError instead of Debug, but do the battle on another PR.

Reply:
Errors go into diags.log, so it's things that the operations people should see.

Reply:
I'm not joining the battle. I'm saying focus on fixing the API issue on this PR.

Reply:
Yahoo is not currently using TSSslSecretGet().  For us, the main purpose of this change was to fix ERRORs in diags.log that we were seeing on config reload.  The change to the API was just part of the effort to prevent future problems as well as the one we were seeing.  Many problems happen first (or only) in prod, and will go unnoticed if they are only seen in debug trace rather than in diags.log.

Reply:
I don't see a reason to do the two things at the same time on one PR.

This line was changed to Debug about two months ago for some reasons. Changing it back to SSLError would need a discussion. I don't want to keep the API problematic until the discussion ends.

Reply:
Do you have a link to that discussion?  If bad configuration files are not an error, what is?'

This is a cleanup PR, so it does many, many things at once.  If you want me to break it out into a large number of small PRs, I'll have to do that in the future as a low-priority task.  Otherwise, Yahoo would fire me for having very very poor judgement about priorities ha ha.

Reply:
The discussion is not started yet. You should start the discussion if you want to change this line back to SSLError. I don't think you should rewrite it as you want without understanding why it was changed to Debug. And the previous change was approved by you. #9225. What was it?

> This is a cleanup PR

I don't understand why you still call this just a clean up. You mean the API is fine as is?

Reply:
@serrislew why did you change this to a debug trace?

Reply:
It's a cleanup PR because it includes many relatively small changes.  One was fixing a potential bug if the TSSslSecretGet() API call was used.  The other changes fixed the config reload error Yahoo was seeing, but I can't identify which change or set of changes constituted the fix.

Reply:
It still calls a SSLError log https://github.com/apache/trafficserver/pull/9225/files#diff-cb865c0bc65fb8ef103a206282b78a50e0c1c9e93ca6713322ba9df9d921e6c9L2453

Reply:
But that's a different line of code.

Reply:
But that additional SSLError log replaces all of the changed SSLError to Debug calls. `SSLPrivateKeyHandler` is only called once and it will reach that additional SSLError when it returns false. And now with the SSLError called after `SSLPrivateKeyHandler` return false, it can print the cert name that wasn't available inside the function

Reply:
Thanks Serris.